### ISSUE TITLE
Better progress when an update runs

### ISSUE THREAD

**Issue by ryneo** [unscramble/jidoteki-admin-api#21]
Currently the `GET /api/v1/admin/update` endpoint responds with a qualitative status, `success`, `failed`, `running`.  Is it possible to get a percentage progress based on the packages or steps that are contained in the update package?

**Comment by aw** [comment]
The update script contains 4 steps, but two of the steps take longer depending on:
- the size of the update package being decrypted
- the size of each individual file to be updated
- the read/write speed of the storage disks

There's no way to guess how long that processing will take, so we would have to somewhat _fake_ the percentages.

I could easily add percentages such as `0%, 25%, 50%, 75%, 100%`, but the time between `25-50%` and between `75%-100%` can vary significantly depending on the factors mentioned above. I'm not entirely sure these percentages will provide any real value if we do that. @ryneo what do you think?

**Comment by aw** [comment]
@ryneo I was thinking about this a bit more, and in fact yes it would be possible to do what you suggest.

I also have some ideas to achieve this without _faking_ anything.

**Comment by ryneo** [comment]
That's awesome!  Looking forward to it.

**Comment by aw** [comment]
It took a while, but this is finally available as of `v1.21.0`. There is a new `percentage` key/value when fetching the status of an update with the `/admin/update` API endpoint, example:

```
HTTP/1.1 200 OK
Content-Type: application/json
{
    "status": "success",
    "percentage": 100,
    "log": "[1432140922][SYSTEM] Updating system successful"
}
```
